api/src/main/java/org/openmrs/module/aijar/dataintegrity/InvalidTBEncounters.java

```diff
@@ -0,0 +1,200 @@
+package org.openmrs.module.aijar.dataintegrity;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hibernate.Query;
+import org.joda.time.DateTime;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.PatientProgram;
+import org.openmrs.Program;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.aijar.metadata.concept.Dictionary;
+import org.openmrs.module.aijar.metadata.core.Programs;
+import org.openmrs.module.dataintegrity.DataIntegrityRule;
+import org.openmrs.module.dataintegrity.rule.RuleResult;
+import org.springframework.stereotype.Component;
+
+/**
+ * Data integrity rules for the TB page which are:
+ * <ol>
+ *  <li>Patients with missing TB identifiers</li>
+ *  <li>Patients with similar TB identifiers</li>
+ *  <li>Patients with no final outcome after 9 months at start of treatment </li>
+ * </ol>
+ */
+@Component
+public class InvalidTBEncounters extends BasePatientRuleDefinition {
+
+	@Override
+	public List<RuleResult<Patient>> evaluate() {
+		List<RuleResult<Patient>> ruleResults = new ArrayList<>();
+		
+		ruleResults.addAll(patientsWithMissingTBNumberIdentifiers("304df0d0-afe4-4a61-a917-d684b100a65a", "Unit TB Number"));
```

Comment:
@moshonk Name the methods properly - patientsWithMissingUnitTBNumber, patientsWithMissingHSDTBNumber, patientsWithMissingDistrictTBNumber ... applies to duplicates also